featuretools/tests/entityset_tests/test_es.py

```diff
@@ -1777,6 +1777,60 @@ def test_entityset_time_type_equality(es):
     assert first_es == second_es
 
 
+def test_entityset_deep_equality(es):
+    first_es = EntitySet()
+    second_es = EntitySet()
+
+    first_es.add_dataframe(dataframe_name='customers',
+                           dataframe=es['customers'].copy(),
+                           index='id',
+                           time_index='signup_date',
+                           logical_types=es['customers'].ww.logical_types,
+                           semantic_tags=get_df_tags(es['customers']))
+
+    second_es.add_dataframe(dataframe_name='sessions',
+                            dataframe=es['sessions'].copy(),
+                            index='id',
+                            logical_types=es['sessions'].ww.logical_types,
+                            semantic_tags=get_df_tags(es['sessions']))
+
+    first_es.add_dataframe(dataframe_name='sessions',
+                           dataframe=es['sessions'].copy(),
+                           index='id',
+                           logical_types=es['sessions'].ww.logical_types,
+                           semantic_tags=get_df_tags(es['sessions']))
+    second_es.add_dataframe(dataframe_name='customers',
+                            dataframe=es['customers'].copy(),
+                            index='id',
+                            time_index='signup_date',
+                            logical_types=es['customers'].ww.logical_types,
+                            semantic_tags=get_df_tags(es['customers']))
```

Comment:
Super minor nitpick, but could we reorder these so we either add the dataframes to the entitysets in the same order or build the first entityset and then the second. I spent some time trying to figure out if there is a reason that we add customers to the first es and then sessions to the second, and then realized it doesn't matter. I found it a little harder to initially follow that we are building equal entitysets with this order.

Reply:
yep, changing to building each entityset at a time.

Reply:
oops, just pushed